src/os_common/os_fhandle.c

```diff
@@ -72,6 +72,33 @@ __wt_handle_is_open(WT_SESSION_IMPL *session, const char *name)
     return (found);
 }
 
+/*
+ * __wt_remove_tiered_local --
+ *     While locked, if the handle is not open, remove the local object.
+ */
+int
+__wt_remove_tiered_local(WT_SESSION_IMPL *session, const char *object, bool *removed)
```

Comment:
I don't think the name of this function should embed the reason we (currently) use it. It's just a function that safely removes a file in a way that avoids races with another thread trying to open the file. We use it for removing local tiered objects today, but we might find other uses for it in the future. 

I would change the name of this function to something more generic. Maybe `__wt_lock_and_remove()`? or `__wt_atomic_remove()`?

Reply:
I like the generic naming suggestions.